src/cdk/bootstrap.ts

```diff
@@ -290,6 +292,8 @@ export class MiraBootstrap {
     }
 
     const cmd = this.args._[0]
+    const cd = new ChangeDetector(process.cwd())
+    const filesChanged = await cd.run()
```

Comment:
there's a small issue that snapshot that is used for the comparison is created in this step, but as CDK runs again later in the code, the modification time is being updated. Effectively it means that deploy will be triggered every time.  

In here https://github.com/nearform/mira/pull/35/files#diff-251cdb280b972bbc33f2c288dc8d52b2R84, the deploy fn was changed to return a promise. When that'll be merged `await cd.takeSnapshot()` should be run after `this.deploy()`. Does it makes sense?

Reply:
well, not sure the exact process, but tested that on the s3 sample app, and deploy was always triggered. Can you check it in the sample app?

Reply:
I tested on sample app, what I noticed now is that `glob` does not preserve file list, so even with same hashes, generated snapshots are different. I'll fix this ordering the array before writing the file, and I think that will fix the issue.